File: connectors-editors/mantis-editor/src/main/scala/com/taskadapter/connector/mantis/editor/MantisProjectsListLoader.scala
package com.taskadapter.connector.mantis.editor

import com.google.common.base.Strings
import com.taskadapter.connector.definition.WebConnectorSetup
import com.taskadapter.connector.definition.exceptions.ServerURLNotSetException
import com.taskadapter.connector.mantis.{MantisManager, MantisManagerFactory, MantisProjectConverter}
import com.taskadapter.model.{GProject, NamedKeyedObject}
import com.taskadapter.web.callbacks.DataProvider

class MantisProjectsListLoader(setup: WebConnectorSetup) extends DataProvider[java.util.List[_ <: NamedKeyedObject]] {
  @throws[ServerURLNotSetException]
  override def loadData(): java.util.List[GProject] = {
    validate(setup)
    val mgr: MantisManager = MantisManagerFactory.createMantisManager(setup)
    try {
      val mntProjects = mgr.getProjects
      new MantisProjectConverter().toGProjects(mntProjects)
    } catch {
      case e: Exception =>
        throw new RuntimeException(e.toString, e)
    }
  }

  @throws[ServerURLNotSetException]
  private def validate(setup: WebConnectorSetup): Unit = {
    if (Strings.isNullOrEmpty(setup.host)) throw new ServerURLNotSetException
  }
}

File: connectors-editors/mantis-editor/src/main/java/com/taskadapter/connector/mantis/editor/MantisProjectsListLoader.java
package com.taskadapter.connector.mantis.editor;

import com.google.common.base.Strings;
import com.taskadapter.connector.definition.WebConnectorSetup;
import com.taskadapter.connector.definition.exceptions.ConnectorException;
import com.taskadapter.connector.definition.exceptions.ServerURLNotSetException;
import com.taskadapter.connector.mantis.MantisManagerFactory;
import com.taskadapter.connector.mantis.MantisProjectConverter;
import com.taskadapter.model.GProject;
import com.taskadapter.model.NamedKeyedObject;
import com.taskadapter.web.callbacks.DataProvider;

import java.util.List;

public class MantisProjectsListLoader implements DataProvider<List<? extends NamedKeyedObject>> {
    private final WebConnectorSetup setup;

    public MantisProjectsListLoader(WebConnectorSetup setup) {
        this.setup = setup;
    }

    @Override
    public List<GProject> loadData() throws ConnectorException {
        validate(setup);
        var mgr = MantisManagerFactory.createMantisManager(setup);
        try {
            var mntProjects = mgr.getProjects();
            return new MantisProjectConverter().toGProjects(mntProjects);
        } catch (Exception e) {
            throw new RuntimeException(e.toString(), e);
        }
    }

    private static void validate(WebConnectorSetup setup) throws ServerURLNotSetException {
        if (Strings.isNullOrEmpty(setup.getHost())) {
            throw new ServerURLNotSetException();
        }
    }
}
